File: SpendWiseRestApi/src/main/java/com/example/demo/entity/PersonalExpenditure.java
package com.example.demo.entity;

import java.sql.Date;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Setter
@Entity
@Table(name = "personalexpenditure")
@JsonIgnoreProperties({"login"}) // Ignore login field to prevent deep recursion
public class PersonalExpenditure {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "exid")
    private int exId;

    @Column(name = "edesc", nullable = false)
    private String eDesc;

    @Column(name = "edate", nullable = false)
    private Date eDate;

    @Column(name = "tid", unique = true, nullable = false)
    private String tId;
    
    @Column(name = "amount",nullable = false)
    private Double amount;

    @ManyToOne
    @JoinColumn(name = "uid")
    private Login login;

    @ManyToOne
    @JoinColumn(name = "pcid")
    @JsonIgnoreProperties("personalExpenditures") // Ignore personalExpenditures in PersonalCategory
    private PersonalCategory personalCategory;
}

File: SpendWiseLoginWebApi/LoginRestApi/Models/Personalexpenditure.cs
using System;
using System.Collections.Generic;

namespace LoginRestApi.Models;

public partial class Personalexpenditure
{
    public int Exid { get; set; }

    public string Edesc { get; set; } = null!;

    public DateOnly Edate { get; set; }

    public string Tid { get; set; } = null!;

    public int? Uid { get; set; }

    public int? Pcid { get; set; }

    public virtual Personalcategory? Pc { get; set; }

    public virtual Login? UidNavigation { get; set; }
}
